pkg/apiserver/api_installer.go

```diff
@@ -216,18 +216,27 @@ func (a *APIInstaller) registerResourceHandlers(path string, storage rest.Storag
 	}
 	versionedStatus := indirectArbitraryPointer(versionedStatusPtr)
 	var (
-		getOptions          runtime.Object
-		versionedGetOptions runtime.Object
-		getOptionsKind      string
-		getSubpath          bool
-		getSubpathKey       string
+		getOptions             runtime.Object
+		versionedGetOptions    runtime.Object
+		getOptionsInternalKind unversioned.GroupVersionKind
+		getOptionsExternalKind unversioned.GroupVersionKind
+		getSubpath             bool
+		getSubpathKey          string
 	)
 	if isGetterWithOptions {
 		getOptions, getSubpath, getSubpathKey = getterWithOptions.NewGetOptions()
-		_, getOptionsKind, err = a.group.Typer.ObjectVersionAndKind(getOptions)
+		getOptionsGVString, getOptionsKind, err := a.group.Typer.ObjectVersionAndKind(getOptions)
```

Comment:
> Sorry for the continuous questions here, but let me ask one more. It seems the only getOptions in our code base is the PodLogOptions, it's handled at path /pods/log, and the reqScope.InternalVersion is "", which is correct. So my question is:
> 
> your reqScope.InternalVersion is for your group, but your getOptions were for /v1` (or unversioned)
> when will this happen?

This ends up mattering downstream when you're trying to have compatibility with other APIs.  The origin /logs endpoints as a for instance.  The final method destination is also shared with `ConnectResource` (the odd refactoring was done as part of the `ListOptions` work of all things).

I think explicitly locating exactly what's needed gives us flexibility where we need it, since query parameters are not currently self-identifying.